app/lib/foreman_deployments/tasks/wait_until_built_task_definition.rb

```diff
@@ -0,0 +1,60 @@
+module ForemanDeployments
+  module Tasks
+    class WaitUntilBuiltTaskDefinition < ForemanDeployments::Tasks::BaseDefinition
+      class Action < BaseAction
+        include Dynflow::Action::Polling
+
+        def done?
+          external_task['built'] == true
+        end
+
+        def timeout
+          input['timeout'] || 2 * 60 * 60 # 2 hours default
+        end
+
+        def invoke_external_task
+          schedule_timeout(timeout) unless timeout <= 0
+          WaitUntilBuiltTaskDefinition.build_status
+        end
+
+        def poll_external_task
+          # TODO: check the case when no host_id is sent
+          host = Host.find(input['host_id'])
+          WaitUntilBuiltTaskDefinition.create_output(host, output)
+          WaitUntilBuiltTaskDefinition.build_status(host)
+        end
+
+        def poll_interval
+          30
+        end
+      end
+
+      def validate
+        ValidationResult.new
+      end
+
+      def preliminary_output(_parameters)
+        WaitUntilBuiltTaskDefinition.create_output(Host::Managed.new)
```

Comment:
It's a bit dangerous to output a new instance of a host as an output for the task. The record is not valid, so if someone will try to validate this model down the line, the validation will fail resulting task validation failure.